nixpkgs/nixpkgs.bzl

```diff
@@ -56,15 +56,26 @@ nixpkgs_local_repository = repository_rule(
     },
 )
 
+def _check_supported_platform(repository_ctx):
+    """
+    Returns:
+       None if the platform is supported. A string error message otherwise.
+    """
+    nix_build = repository_ctx.which("nix-build")
+    if nix_build != None:
+        return None
+    return "Platform is not supported: nix-build not found in PATH. See attribute fail_not_supported if you don't want to use Nix."
```

Comment:
I'm not fond of this approach. This code is surprising and thereby confusing in a couple ways:
* It returns `None` for a positive result
* It returns a non-empty string for a negative result
* It kinda overloads the error message string to function as a boolean at the same time.

I also see no reason for this. The call `repository_ctx.which("nix-build")` is already factored out into `_is_supported_platform`, no reason to factor it out further. The error message is only used in one place, so there's no benefit to factor it out, and even if, the error message is constant, so you could just factor it out into a global binding.

I suggest the following:
* Roll back this change.
* Just edit the error message passed to `fail`.

Reply:
I get your point, though in a larger codebase I wouldn't have an error message 50 lines away from the corresponding check, hence my previous solution. But it's also overkill in the current state.

I've edited the commit accordingly.

Reply:
> I get your point, though in a larger codebase I wouldn't have an error message 50 lines away from the corresponding check, hence my previous solution.

That's a fair point, but in that case I'd go for a solution with less complexity. Maybe something along the lines of

```
def _get_nix_build_or_fail(repository_ctx, fail_not_supported):
    nix_build = repository_ctx.which("nix-build")
    if nix_build == None and fail_not_supported:
        fail(...)
    else:
        return nix_build

def _nixpkgs_package_impl(repository_ctx):
    ...
    nix_build_path = _get_nix_build_or_fail(...)
    if nix_build_path == None:
        return
    ...
```